src/Item.js

```diff
@@ -0,0 +1,48 @@
+import React, { useState, useEffect } from 'react';
+import { db } from './lib/firebase';
+
+function Item(props) {
+  const [checked, setChecked] = useState(false);
+
+  useEffect(() => {
+    // setChecked(props.lastPurchaseDate && true);
+    if (props.lastPurchaseDate) {
+      const timeChecked = 86400; //Change this value if you don't want to wait 24 hrs
+      const timeElapsed = Date.now() / 1000 - props.lastPurchaseDate.seconds;
+      setChecked(timeElapsed <= timeChecked && true);
+    }
+  }, [props.lastPurchaseDate]);
+
+  const handleClick = () => {
+    if (checked) {
+      db.collection(props.userToken).doc(props.id).update({
+        'formData.lastPurchaseDate': null,
+      });
```

Comment:
I don't think you want to set the lastPurchaseDate to `null` here. In future issues you will be calculating the next estimated purchase date based on your previous purchases. So, let's say for example you purchased cheese on 4/24/2021, but then unchecked it and set that value to null. You wouldn't have that 4/24/2021 purchase date available to calculate your next estimated purchase date. 

https://github.com/the-collab-lab/tcl-22-smart-shopping-list/issues/11

Reply:
Oh, good point @laurenmbeatty! Our thinking here was to change it back to `null` if the user were to manually uncheck the box. So based on what you're saying, setting the date to null only works if this were the first time the user clicked on the item as purchased but then wanted to uncheck manually for some reason (eg. accidentally clicked). If it's an item that's been purchase before, we want to have the date that was there prior to the latest date that's been added due to an accidental click. Totally makes sense.

Reply:
Latest change seems to have resolved this!